From bbcecc404e5fad4972eca627374dcafbc8011c94 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Wed, 25 Jun 2014 11:31:43 +1000
Subject: [PATCH 01/16] Fix PiP when using OpenGL output

Viewport must be of the output window, not of the main video currently playing
---
 mythtv/libs/libmythtv/openglvideo.cpp     |    2 +-
 mythtv/libs/libmythtv/videoout_opengl.cpp |    6 ++----
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/mythtv/libs/libmythtv/openglvideo.cpp b/mythtv/libs/libmythtv/openglvideo.cpp
index dbd39e2..2de7d4c 100644
--- a/mythtv/libs/libmythtv/openglvideo.cpp
+++ b/mythtv/libs/libmythtv/openglvideo.cpp
@@ -141,7 +141,7 @@ bool OpenGLVideo::Init(MythRenderOpenGL *glcontext, VideoColourSpace *colourspac
     display_visible_rect  = displayVisibleRect;
     display_video_rect    = displayVideoRect;
     video_rect            = videoRect;
-    masterViewportSize    = QSize(1920, 1080);
+    masterViewportSize    = display_visible_rect.size();
     frameBufferRect       = QRect(QPoint(0,0), video_disp_dim);
     softwareDeinterlacer  = "";
     hardwareDeinterlacing = false;
diff --git a/mythtv/libs/libmythtv/videoout_opengl.cpp b/mythtv/libs/libmythtv/videoout_opengl.cpp
index 267f844..76af4c3 100644
--- a/mythtv/libs/libmythtv/videoout_opengl.cpp
+++ b/mythtv/libs/libmythtv/videoout_opengl.cpp
@@ -872,8 +872,7 @@ void VideoOutputOpenGL::ShowPIP(VideoFrame  *frame,
                      dvr, position,
                      QRect(0, 0, pipVideoWidth, pipVideoHeight),
                      false, options, false);
-        QSize viewport = gl_videochain ? gl_videochain->GetViewPort() :
-                                         window.GetDisplayVisibleRect().size();
+        QSize viewport = window.GetDisplayVisibleRect().size();
         gl_pipchain->SetMasterViewport(viewport);
         if (!success)
         {
@@ -898,8 +897,7 @@ void VideoOutputOpenGL::ShowPIP(VideoFrame  *frame,
             QRect(0, 0, pipVideoWidth, pipVideoHeight),
             false, options, false);
 
-        QSize viewport = gl_videochain ? gl_videochain->GetViewPort() :
-                                         window.GetDisplayVisibleRect().size();
+        QSize viewport = window.GetDisplayVisibleRect().size();
         gl_pipchain->SetMasterViewport(viewport);
 
         if (!success)
-- 
1.7.10.2

